### PR TITLE
basic langchain provider implementation

### PR DESCRIPTION
**Pull request by Nvillaluenga**
**Addressing this issue:** https://github.com/truera/trulens/issues/645

Adding very basic support to use langchain LLMs and ChatModels as provider in the feedback module

Notes:
- Made smoke test with answer relevance and it runs.
  - It does not work perfectly, latency is not working for example (which makes sense because I did nothing for it to work)

### CODE REVIEW

**Comment by joshreini1** [comment]
Thanks @Nvillaluenga for the contribution!

I'll take a look at what's needed to latency tracking here.

File: trulens_eval/trulens_eval/feedback/provider/endpoint/langchain.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    def handle_generation(self, response: Any) -> None:
+        super().handle_generation(response)
+
+
+class LangchainEndpoint(Endpoint, WithClassInfo):
+    """
+    Langchain endpoint.
+    """
+
+    client: Union[BaseLLM, BaseChatModel]
```
Comment (piotrm0) [inline trulens_eval/trulens_eval/feedback/provider/endpoint/langchain.py:31]: Rename to "chain" or something relevant to what BaseLLM/BaseChatModel are.

File: trulens_eval/trulens_eval/feedback/provider/langchain.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        from langchain.llms import OpenAI
+
+        gpt3_llm = OpenAI(openai_api_key=OPENAI_API_KEY, model="gpt-3.5-turbo-instruct")
+        langchain_provider = Langchain(client = client)
+        ```
+
+        Args:
+            client (Union[BaseLLM, BaseChatModel]): Langchain LLMs or chat models
+        """
+        self_kwargs = kwargs.copy()
```
Comment (piotrm0) [inline trulens_eval/trulens_eval/feedback/provider/langchain.py:41]: Make a separate `kwargs` for `LangchainEndpoint` and for self; pass in only what they each need.

File: trulens_eval/trulens_eval/feedback/provider/endpoint/langchain.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        bindings: inspect.BoundArguments,
+        response: Any,
+        callback: Optional[EndpointCallback],
+    ) -> None:
+        # TODO: Implement this and wrapped
+        self.global_callback.handle_generation(response=None)
+        if callback is not None:
+            callback.handle_generation(response=None)
+
+    def __init__(self, *args, **kwargs):
```
Comment (piotrm0) [inline trulens_eval/trulens_eval/feedback/provider/endpoint/langchain.py:48]: If `client` (or renamed as 'chain`), must be given, make it an arg here.

File: trulens_eval/trulens_eval/feedback/provider/langchain.py (hunk 2 above)
Comment (Nvillaluenga) [reply trulens_eval/trulens_eval/feedback/provider/langchain.py:41]: Done

**Review by joshreini1 — APPROVED** [review]

**Comment by zeroasterisk** [comment]
Any chance this could be added to docs? 

I am looking for the ability to use other models (not provided by OpenAI) and also #789 looks like it is also asking for similar features... I'm pretty sure this PR unlocks that ability, but it isn't clearly explained how to do so.

Can I request a bit of guidance or example on how to integrate other LLM providers for feedback?

**Comment by joshreini1** [comment]
Hey @zeroasterisk - this is already in docs. See: https://www.trulens.org/trulens_eval/api/langchain_provider/

**Pull request merged**